src/Reader/Reader.php

```diff
@@ -117,23 +118,30 @@ public static function setCache(CacheStorage $cache)
      *
      * Sets the HTTP client object to use for retrieving the feeds.
      *
-     * @param  ZendHttp\Client $httpClient
+     * @param  ZendHttp\Client | Http\ClientInterface $httpClient
      * @return void
      */
-    public static function setHttpClient(ZendHttp\Client $httpClient)
+    public static function setHttpClient($httpClient)
     {
+        if ($httpClient instanceof ZendHttp\Client) {
+            $httpClient = new Http\ZendHttpClientDecorator($httpClient);
+        }
+
+        if (! $httpClient instanceof Http\ClientInterface) {
```

Comment:
I don't understand the point you're trying to make. The code is written such that a client doesn't need to expose headers, for BC purposes. The check here is to ensure that we have a client implementation, period.